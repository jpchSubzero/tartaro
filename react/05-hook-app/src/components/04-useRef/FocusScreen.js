import React, { useRef } from 'react';

import './styles.css';

export const FocusScreen = () => {

    // Agregar useRef luego de probar el select()
    const inputNameRef = useRef();
    console.log(inputNameRef);

    const handleClick = () => {
        // Permite darle focus al elemento seleccionado
        // document.querySelector('#name').focus();

        // Permite darle focus al elemento seleccionado pero tambi√©n seleccionar todo el contenido por lo que resulta mejor que simplemente focus
        // document.querySelector('#name').select();

        // Usamos el inputNameRef para referirnos al elemento directamente
        inputNameRef.current.select();
        console.log(inputNameRef);
    };

    return (
        <>
            <h1>Focus screen</h1>
            <hr />

            <input 
                // Usamos el inputNameRef para crear una referencia al elemento
                ref={inputNameRef}
                id="name"
                className="form-control mb-3" 
                placeholder="Ingrese un nombre"
                type="text" 
            />
            <input 
                id="email"
                className="form-control mb-3" 
                placeholder="Ingrese un email"
                type="email" 
            />

            <button
                className="btn btn-outline-primary"
                onClick={handleClick}
            >Focus</button>
        </>
    )
}
