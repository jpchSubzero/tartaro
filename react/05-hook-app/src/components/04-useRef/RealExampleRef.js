import React, { useState } from 'react';
import './styles.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleRef = () => {

    // Agregar el state para controlar la visibilidad del componente MultipleCustomHooks
    const [show, setShow] = useState(false);

    return (
        <>
            <h1>RealExampleRef</h1>  
            <hr />

            {/* Agregar un botón para hacer el toogle del componente MultipleCustomHooks */}
            {/* Luego del toogle agregar el setTimeOut al hook useFetch para hacer que demore */}
            {/* Para generar el error hay que mostrar la frase, solicitar la siguiente y ahí mismo ocultar; esto causa que el componente sea desmontado pero el proceso fetch sigue ejecutándose y al concluir ya no tiene un componente a quien devolver la respuesta */}
            {/* Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function. */}
            <button className="btn btn-primary" onClick={() => setShow(!show)}>Mostrar/ocultar</button>

            {/* <MultipleCustomHooks /> */}

            {/* Controlar el componente a través del state show */}
            { show && <MultipleCustomHooks /> }
        </>
    )
}
