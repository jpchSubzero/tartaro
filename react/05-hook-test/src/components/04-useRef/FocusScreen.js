import React, { useRef } from 'react';

import './styles.css';

export const FocusScreen = () => {

    const inputNameRef = useRef();

    const handleClick = () => {
        inputNameRef.current.select();
    };

    return (
        <>
            <h1>Focus screen</h1>
            <hr />

            <input 
                ref={inputNameRef}
                id="name"
                className="form-control mb-3" 
                placeholder="Ingrese un nombre"
                type="text" 
            />
            <input 
                id="email"
                className="form-control mb-3" 
                placeholder="Ingrese un email"
                type="email" 
            />

            <button
                className="btn btn-outline-primary"
                onClick={handleClick}
            >Focus</button>
        </>
    )
}
