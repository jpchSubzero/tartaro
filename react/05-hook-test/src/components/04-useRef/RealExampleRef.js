import React, { useState } from 'react';
import './styles.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <>
            <h1>RealExampleRef</h1>  
            <hr />

            <button className="btn btn-primary" onClick={() => setShow(!show)}>Mostrar/ocultar</button>

            { show && <MultipleCustomHooks /> }
        </>
    )
}
