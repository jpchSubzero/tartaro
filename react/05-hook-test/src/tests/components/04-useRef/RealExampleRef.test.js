import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Prueba de RealExampleRef', () => {
    const wrapper = shallow(<RealExampleRef />);

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });
    
    test('Debe mostrar el componenete MultipleCustomHooks', () => {
        wrapper.find('button').simulate('click');        
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

        wrapper.find('button').simulate('click');        
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });
});

// Tarea 1
// Debe mostrarse correctamente
// Debe mostrar el componenete MultipleCustomHooks