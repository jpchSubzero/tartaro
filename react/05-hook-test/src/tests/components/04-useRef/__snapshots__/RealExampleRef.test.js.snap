// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de RealExampleRef Debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Mostrar/ocultar
  </button>
</Fragment>
`;
